src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -49,7 +50,7 @@ public function init($redisClient, $namespace = '', $defaultLifetime = 0)
         }
         if ($redisClient instanceof \RedisCluster) {
             $this->enableVersioning();
-        } elseif (!$redisClient instanceof \Redis && !$redisClient instanceof \RedisArray && !$redisClient instanceof \Predis\Client) {
+        } elseif (!$redisClient instanceof \Redis && !$redisClient instanceof \RedisArray && !$redisClient instanceof \Predis\Client && !$redisClient instanceof RedisProxy) {
             throw new InvalidArgumentException(sprintf('%s() expects parameter 1 to be Redis, RedisArray, RedisCluster or Predis\Client, %s given', __METHOD__, is_object($redisClient) ? get_class($redisClient) : gettype($redisClient)));
```

Comment:
this message needs an update right? Same for below.

Reply:
I'd prefer not: RedisProxy is internal (same for docblocks)

Reply:
right! :+1: